lib/src/node/value/calculation.dart

```diff
@@ -11,33 +11,24 @@ import '../../value.dart';
 import '../reflection.dart';
 
 /// Check that [arg] is a valid argument to a calculation function.
-void assertCalculationValue(Object arg, {bool checkUnquoted = true}) {
+void assertCalculationValue(Object arg) {
   if (arg is! SassNumber &&
       arg is! SassString &&
       arg is! SassCalculation &&
       arg is! CalculationOperation &&
       arg is! CalculationInterpolation) {
-    jsThrow(JsError('Argument must be one of '
+    jsThrow(JsError('Argument `$arg` must be one of '
         'SassNumber, SassString, SassCalculation, CalculationOperation, '
         'CalculationInterpolation'));
   }
-  if (checkUnquoted && arg is SassString && arg.hasQuotes) {
-    jsThrow(JsError('Argument must be unquoted SassString'));
-  }
-  if (arg is CalculationOperation) {
-    assertCalculationValue(arg.left, checkUnquoted: checkUnquoted);
-    assertCalculationValue(arg.right, checkUnquoted: checkUnquoted);
+  if (arg is SassString && arg.hasQuotes) {
```

Comment:
I've been trying out a more pattern-matching style lately:

```suggestion
  if (arg case SassString(hasQuotes: true)) {
```

It's a little longer here, but only because the repeated variable name was so short, and I think it's a bit easier to understand.